File: ruby-elasticemail-examples/examples/email_status.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

##
# Email Status
#
# Both ids are returned by every send call.
#
# Usage: ruby examples/email_status.rb <transactionId> [messageId]

require_relative "ee"

emails_api = ElasticEmail::EmailsApi.new

transaction_id, message_id = ARGV

unless transaction_id
  warn "Usage: ruby examples/email_status.rb <transactionId> [messageId]"
  exit 1
end

begin
  data = emails_api.emails_by_transactionid_status_get(
    transaction_id,
    show_failed: true,
    show_sent: true,
    show_delivered: true,
    show_pending: true,
    show_opened: true,
    show_clicked: true
  )

  puts "=== Transaction status ==="
  puts "Status:      #{data.status}"
  puts "Recipients:  #{data.recipients_count}"
  puts "Sent:        #{data.sent_count} #{(data.sent || []).inspect}"
  puts "Delivered:   #{data.delivered_count} #{(data.delivered || []).inspect}"
  puts "Pending:     #{data.pending_count}"
  puts "Opened:      #{data.opened_count}"
  puts "Clicked:     #{data.clicked_count}"
  puts "Failed:      #{data.failed_count} #{(data.failed || []).map(&:to_hash).inspect}"
rescue ElasticEmail::ApiError => e
  api_error("fetch status", e)
end

if message_id
  begin
    data = emails_api.emails_by_msgid_view_get(message_id)
    puts "\n=== Message ==="
    puts "From:     #{data.preview&.from}"
    puts "Subject:  #{data.preview&.subject}"
    puts "Status:   #{data.status&.status_name} #{data.status&.date_sent}"
    body = data.preview&.body.to_s
    puts "Body preview: #{body.length > 200 ? "#{body[0, 200]}..." : body}"
  rescue ElasticEmail::ApiError => e
    warn "Error fetching message: #{e.code} #{e.response_body || e.message}"
  end
end

File: hono-elasticemail-examples/javascript/examples/email-status.js
import "dotenv/config";
import { Configuration, EmailsApi } from "@elasticemail/elasticemail-client-ts-axios";

const config = new Configuration({ apiKey: process.env.ELASTICEMAIL_API_KEY });
const emailsApi = new EmailsApi(config);

// Usage: node examples/email-status.js <transactionId> [messageId]
// Both ids are returned by every send call.
const [transactionId, messageId] = process.argv.slice(2);

if (!transactionId) {
  console.error("Usage: node examples/email-status.js <transactionId> [messageId]");
  process.exit(1);
}

try {
  const { data } = await emailsApi.emailsByTransactionidStatusGet(
    transactionId,
    true, // showFailed
    true, // showSent
    true, // showDelivered
    true, // showPending
    true, // showOpened
    true, // showClicked
  );

  console.log("=== Transaction status ===");
  console.log("Status:     ", data.Status);
  console.log("Recipients: ", data.RecipientsCount);
  console.log("Sent:       ", data.SentCount, data.Sent ?? []);
  console.log("Delivered:  ", data.DeliveredCount, data.Delivered ?? []);
  console.log("Pending:    ", data.PendingCount);
  console.log("Opened:     ", data.OpenedCount);
  console.log("Clicked:    ", data.ClickedCount);
  console.log("Failed:     ", data.FailedCount, data.Failed ?? []);
} catch (err) {
  console.error("Error fetching status:", err.response?.status, err.response?.data ?? err.message);
  process.exit(1);
}

if (messageId) {
  try {
    const { data } = await emailsApi.emailsByMsgidViewGet(messageId);
    console.log("\n=== Message ===");
    console.log("From:    ", data.Preview?.From);
    console.log("Subject: ", data.Preview?.Subject);
    console.log("Status:  ", data.Status?.StatusName, data.Status?.DateSent ?? "");
    const body = data.Preview?.Body ?? "";
    console.log("Body preview:", body.length > 200 ? body.slice(0, 200) + "..." : body);
  } catch (err) {
    console.error("Error fetching message:", err.response?.status, err.response?.data ?? err.message);
  }
}
